From cea09d48c9da6bb7cb441527b5ab95e064718479 Mon Sep 17 00:00:00 2001
From: Alexandru Costache <alexandru@balena.io>
Date: Fri, 28 Aug 2020 14:52:27 +0200
Subject: [PATCH] rp4: Fix USB boot on Pi4 8GB

Without this patch, u-boot will fail to load
and reboot continuously on 8GB Pi4 when booting
from USB.

See: https://patchwork.ozlabs.org/project/linux-pci/patch/20200629161845.6021-5-nsaenzjulienne@suse.de/

Upstream-status: pending
Signed-off-by: Alexandru Costache <alexandru@balena.io>
---
 arch/arm/boot/dts/bcm2711-rpi-4-b.dts | 12 ++++++++++--
 1 file changed, 10 insertions(+), 2 deletions(-)

diff --git a/arch/arm/boot/dts/bcm2711-rpi-4-b.dts b/arch/arm/boot/dts/bcm2711-rpi-4-b.dts
index 0c4abbdc794c..513cae21e64c 100644
--- a/arch/arm/boot/dts/bcm2711-rpi-4-b.dts
+++ b/arch/arm/boot/dts/bcm2711-rpi-4-b.dts
@@ -198,9 +198,17 @@
 };
 
 &pcie0 {
-	usb@1,0 {
+	pci@1,0 {
+		#address-cells = <3>;
+		#size-cells = <2>;
+		ranges;
+
 		reg = <0 0 0 0 0>;
-		resets = <&reset RASPBERRYPI_FIRMWARE_RESET_ID_USB>;
+
+		usb@1,0 {
+			reg = <0x10000 0 0 0 0>;
+			resets = <&reset RASPBERRYPI_FIRMWARE_RESET_ID_USB>;
+		};
 	};
 };
 
-- 
2.17.1

